File: src/de/schauderhaft/cleanCode/exercise6/InstanceClassifier.java
package de.schauderhaft.cleanCode.exercise6;

import static java.util.Arrays.asList;

import java.util.Collection;

import de.schauderhaft.cleanCode.stubs.SrvParameterCache;

public class InstanceClassifier {

    private final Collection<String> testInstances = asList("grglmrpft",
            "sysmrpt", "gru23b12", "gru23b13");
    private final Collection<String> qaInstances = asList("grglmrpfqs",
            "gru23c12", "sysglmqs", "sysgmrpf-qs");

    boolean isTestInstance() {
        return matchesAny(sanatizedServerUrl(), testInstances);
    }

    boolean isQaInstance() {
        return matchesAny(sanatizedServerUrl(), qaInstances);
    }

    boolean isProductionInstance() {
        return !(isTestInstance() || isQaInstance());
    }

    private boolean matchesAny(String sanitizedUrl, Collection<String> instances) {
        for (String pattern : instances) {
            if (sanitizedUrl.contains(pattern))
                return true;
        }
        return false;
    }

    private String sanatizedServerUrl() {
        String url = SrvParameterCache
                .getParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
        if (url == null)
            url = "";

        return url.toLowerCase();
    }
}

File: src/de/schauderhaft/cleanCode/exercise6/InstanceClassifier.cs
using System;
using System.Linq;
using System.Collections.Generic;
using de.schauderhaft.cleanCode.stubs;

namespace de.schauderhaft.cleanCode.exercise6
{
    public class InstanceClassifier
    {
        private readonly IEnumerable<string> testInstances = new List<string> 
        {"grglmrpft", "sysmrpt", "gru23b12", "gru23b13"};
        
        private readonly IEnumerable<string> qaInstances = new List<string>
        { "grglmrpfqs", "gru23c12", "sysglmqs", "sysgmrpf-qs"};
        
        public bool IsTestInstance()
        {
            return testInstances.Contains(SanatizedServerUrl());
        }
        
        public bool IsQaInstance()
        {
            return qaInstances.Contains(SanatizedServerUrl());
        }
        
        public bool IsProductionInstance()
        {
            return !(IsTestInstance() || IsQaInstance());
        }
        
        private string SanatizedServerUrl() 
        {
            string url = SrvParameterCache
                .GetParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
            if (url == null)
                url = string.Empty;

            return url.ToLower();
        }
    }
}